Dockerfile

```diff
@@ -0,0 +1,55 @@
+FROM debian:bullseye-slim as builder
+
+#Build dependencies
+ARG DEBIAN_FRONTEND=noninteractive
+RUN apt-get update
+RUN apt-get install -y  build-essential libtool autotools-dev automake pkg-config  bsdmainutils  git unzip wget curl python3-git cargo clang
+
+COPY . /bitcoinunlimited
+
+#Build
+WORKDIR /bitcoinunlimited
+
+#Prefetch from alternate URL because the mirror hardcoded in depends is down
+#TODO: Maybe we should download and keep all the tars in one place
+RUN mkdir -p /bitcoinunlimited/depends/sources && curl -Lo /bitcoinunlimited/depends/sources/boost_1_66_0.tar.bz2 https://boostorg.jfrog.io/artifactory/main/release/1.66.0/source/boost_1_66_0.tar.bz2
+#Static compile
+#https://gitlab.com/bitcoinunlimited/BCHUnlimited/-/blob/dev/doc/build-unix.md#produce-static-binaries
+RUN cd depends && make -j$(grep -c ^processor /proc/cpuinfo) NO_QT=1  && cd ..
+
+RUN ./autogen.sh
+RUN ./configure --prefix=$PWD/depends/x86_64-pc-linux-gnu  --with-gui=no
```

Comment:
This seems to have alot less argument than. The upstream one. https://gitlab.com/bitcoinunlimited/BCHUnlimited/-/blob/dev/docker/Dockerfile.ubuntu18-linux32#L24 is it not need? something like `--disable-dependency-tracking`

Reply:
Interesting, I did not reference the upstream dockerfile enough. Let me review what the flag does and test. I will try to get our dockerfile as close as possible to the upstream. I followed the guide from https://gitlab.com/bitcoinunlimited/BCHUnlimited/-/blob/dev/doc/build-unix.md#produce-static-binaries

Reply:
The referenced dockerfile is very old and likely not maintained. It also does not build/install electrscash . I wouldnt look too closely at that.

I added 
- `--disable-dependency-tracking` - "speeds up one-time build"
- `--enable-reduce-exports` - "attempt to reduce exported symbols in the resulting executables (default is no)"

I skipped
- `--enable-glibc-back-compat` since we don't care about glibc compatibility for our executables as they are only intended for our docker image
- `LDFLAGS=-static-libstdc++` Our build works just fine without it.
- `--with-incompatible-bdb` I think this is not really needed since the depends system is downloading and compiling the compatible bdb version